File: docs/specs/golden/syntax/csharp/basics.cs
// Conformance case for the C# syntax extractor (plugin-interface.md, sections 3 to 5).
using System;
using System.Collections.Generic;

namespace Shop.Core;

/// <summary>An order.</summary>
public partial class Order : IPrintable
{
    public const int MaxItems = 50;
    private readonly List<string> _items = new List<string>();
    public event EventHandler Changed;
    public string Id { get; init; } = "";
    public string this[int index] => _items[index];

    public Order() { }
    static Order() { }
    ~Order() { }

    /// <summary>Adds an item.</summary>
    public bool Add(string item)
    {
        if (item.Length == 0 || _items.Count >= MaxItems)
        {
            return false;
        }
        _items.Add(item);
        Changed.Invoke(this, EventArgs.Empty);
        return true;
    }

    public bool Add(string item, int quantity)
    {
        for (var i = 0; i < quantity; i++)
        {
            Add(item);
        }
        return true;
    }

    public T Get<T>(int index) => (T)(object)_items[index];

    public static explicit operator string(Order order) => order.Id;

    void IPrintable.Print() => Console.WriteLine(Id);

    public enum State { Draft, Placed }
}

public interface IPrintable { void Print(); }

public record Customer(string Name, int Age);

public delegate void Notify(Order order);

File: docs/specs/golden/syntax/python/basics.py
"""Conformance case for the Python syntax extractor (plugin-interface.md, sections 3 to 5)."""

import logging
from typing import overload

from .models import Order

MAX_ITEMS = 50
logger = logging.getLogger(__name__)


class Cart:
    """A shopping cart."""

    currency = "EUR"

    def __init__(self, owner: str) -> None:
        self.owner = owner
        self._items = []

    @property
    def size(self) -> int:
        return len(self._items)

    @size.setter
    def size(self, value: int) -> None:
        raise NotImplementedError

    @staticmethod
    def empty() -> "Cart":
        return Cart("")

    def add(self, item: str, quantity: int = 1) -> bool:
        if not item or quantity < 1:
            return False
        for _ in range(quantity):
            self._items.append(item)
        return True


@overload
def total(values: list) -> int: ...
@overload
def total(values: tuple) -> float: ...
def total(values):
    return sum(v for v in values if v)


if __name__ == "__main__":
    logger.info(total([1, 2]))
